### PR TITLE
Fix issues identified by npm audit

### PR DESCRIPTION
**Pull request by ianoxley**
Fix issues highlighted by running `npm audit`.

Require `npm@6`.

**NB** gulp will not be updated, as it will soon be replaced with webpack.

## Related Issues / Pull Requests
#1746 
https://github.com/Sage/carbon-factory/pull/194

## TODO
- [x] Update `carbon-factory` to non-build branch

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,14 +31,14 @@
   },
   "devDependencies": {
     "babel-standalone": "~6.17.0",
-    "carbon-factory": "3.3.3",
+    "carbon-factory": "git+ssh://git@github.com/Sage/carbon-factory.git#build-npm-6-audit",
     "enzyme": "^3.3.0",
     "enzyme-adapter-react-16": "^1.1.1",
-    "express": "~4.14.0",
+    "express": "^4.16.3",
```
Comment (andrewjtait) [inline package.json]: would we not prefer tilde?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
+# 4.0.0
```
Comment (andrewjtait) [inline CHANGELOG.md]: Is this a breaking change?

Comment (ianoxley) [reply CHANGELOG.md]: It's dependent on #1746 to get rid of the gulp issue identified by `npm audit`, hence why I'm targeting `v4.0.0`.

Comment (andrewjtait) [reply CHANGELOG.md]: Ah ok - FYI we will not lose gulp immediately as we still need it for legacy lint tasks :( however I don't think this poses any risk

File: package.json [hunk 3]
```diff
@@ -31,14 +31,14 @@
...
+    "carbon-factory": "git+ssh://git@github.com/Sage/carbon-factory.git#build-npm-6-audit",
     "enzyme": "^3.3.0",
     "enzyme-adapter-react-16": "^1.1.1",
-    "express": "~4.14.0",
+    "express": "~4.16.3",
     "flux": "^3.1.1",
     "gulp": "~3.9.0",
     "highlight.js": "~9.6.0",
-    "jest": "~20.0.4",
+    "jest": "^22.4.3",
```
Comment (andrewjtait) [inline package.json]: im not sure why we even have this - it comes from carbon factory

File: package.json [hunk 4]
```diff
@@ -31,14 +31,14 @@
   },
   "devDependencies": {
     "babel-standalone": "~6.17.0",
-    "carbon-factory": "3.3.3",
+    "carbon-factory": "git+ssh://git@github.com/Sage/carbon-factory.git#build-npm-6-audit",
     "enzyme": "^3.3.0",
     "enzyme-adapter-react-16": "^1.1.1",
-    "express": "~4.14.0",
+    "express": "~4.16.3",
```
Comment (andrewjtait) [inline package.json]: dont need this for webpack

File: CHANGELOG.md (hunk 2 above)
Comment (andrewjtait) [reply CHANGELOG.md]: 4.1.0?

**Comment by abramin** [comment]
QAd in demo site - no new console errors, checked most components, guide pages.

**Pull request merged**